src/client.rs

```diff
@@ -748,8 +735,29 @@ impl Client {
             .send()
             .await?;
 
-        trace!(headers=?res.headers(), "Got Headers");
-        if res.headers().get("Docker-Content-Digest").is_none() {
+        if let Some(digest) = digest_header_value(res.headers().clone())? {
+            let status = res.status();
+            let body = res.bytes().await?;
+            validate_registry_response(status, &body, &url)?;
+
+            // If the reference has a digest and the digest header has a matching algorithm, compare
```

Comment:
If the reference has a digest and the digest header has a different algorithm, the returned digest is not verified. Is that intentional?

Reply:
Yes, because the behavior here is unclear. We are only `HEAD`-ing the manifest here and returning the digest in the header if it exists. Perhaps the proper behavior of this function is to only ever fetch the complete manifest, which we could definitely change in the next version

Reply:
Makes sense, and there is probably no good use case for fetching the digest of a manifest by digest :)